edward/inferences/hmc.py

```diff
@@ -152,18 +152,23 @@ def _log_joint(self, z_sample):
     return log_joint
 
 
-def leapfrog(z_old, r_old, step_size, log_joint):
+def leapfrog(z_old, r_old, step_size, log_joint, n_steps):
   z_new = {}
   r_new = {}
 
+  for key in z_old:
+    z_new[key] = z_old[key]
+    r_new[key] = r_old[key]
+
   grad_log_joint = tf.gradients(log_joint(z_old), list(six.itervalues(z_old)))
-  for i, key in enumerate(six.iterkeys(z_old)):
-    z, r = z_old[key], r_old[key]
-    r_new[key] = r + 0.5 * step_size * grad_log_joint[i]
-    z_new[key] = z + step_size * r_new[key]
-
-  grad_log_joint = tf.gradients(log_joint(z_new), list(six.itervalues(z_new)))
-  for i, key in enumerate(six.iterkeys(z_old)):
-    r_new[key] += 0.5 * step_size * grad_log_joint[i]
+  for n in range(n_steps):
```

Comment:
since we're not using the looping variable, we should do `for _ in ...` for readability